### PR TITLE
Deterministic Python Testing

### PR DESCRIPTION
**Pull request by christophernorth**
Python tests should now all automatically be deterministic (ie start with a known random seed and vertex ID values)

### CODE REVIEW

File: keanu-project/src/main/java/io/improbable/keanu/vertices/VertexId.java [hunk 1]
```diff
@@ -84,4 +84,8 @@ public int getIndentation() {
     public long[] getValue() {
         return Arrays.copyOf(idValues, idValues.length);
     }
+
+    public static void resetIdGenerator() {
+        ID_GENERATOR.set(0);
```
Comment (GeorgeNash) [inline keanu-project/src/main/java/io/improbable/keanu/vertices/VertexId.java]: Were we setting this to `1` for any particular reason before?

File: keanu-python/keanu/keanu_random.py [hunk 2]
```diff
@@ -22,3 +23,8 @@ def next_double(self) -> float:
     @staticmethod
     def set_default_random_seed(seed: int) -> None:
         k.jvm_view().KeanuRandom.setDefaultRandomSeed(seed)
+
+
+def set_deterministic_state():
```
Comment (GeorgeNash) [inline keanu-python/keanu/keanu_random.py]: should this be a static method?

File: keanu-python/tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import pytest
+
+from keanu import set_deterministic_state
+
+
+@pytest.fixture(autouse=True)
```
Comment (GeorgeNash) [inline keanu-python/tests/conftest.py]: nice

File: keanu-python/keanu/keanu_random.py [hunk 4]
```diff
@@ -22,3 +23,8 @@ def next_double(self) -> float:
...
+    KeanuRandom.set_default_random_seed(0)
+    k.jvm_view().VertexId.resetIdGenerator()
```
Comment (GeorgeNash) [inline keanu-python/keanu/keanu_random.py]: can we write a quick test to make sure it resets the Id as expected? Or have we got one on the Java side that covers it

File: keanu-project/src/main/java/io/improbable/keanu/vertices/VertexId.java (hunk 1 above)
Comment (christophernorth) [reply keanu-project/src/main/java/io/improbable/keanu/vertices/VertexId.java]: Not really - it was just a good known value - I've decided to set it 0 everywhere just as that's the value we start at normally so it seems more consistent.

File: keanu-python/keanu/keanu_random.py (hunk 2 above)
Comment (christophernorth) [reply keanu-python/keanu/keanu_random.py]: I personally think it's better just being a function given it's not specifically related to the KeanuRandom object - it's just something that stops things being non-deterministic and interfaces with the KeanuRandom class to do that.  Happy to hear more thoughts though.

File: keanu-python/keanu/keanu_random.py (hunk 4 above)
Comment (christophernorth) [reply keanu-python/keanu/keanu_random.py]: Good suggestion - I'll knock up a super quick test

File: keanu-python/tests/conftest.py (hunk 3 above)
Comment (christophernorth) [reply keanu-python/tests/conftest.py]: 👍

**Pull request merged**